addons/purchase_stock/models/stock_rule.py

```diff
@@ -123,13 +124,40 @@ def _run_buy(self, procurements):
                         procurement.values, po))
             self.env['purchase.order.line'].sudo().create(po_line_values)
 
+    def _get_lead_days(self, product):
+        """Extend the fonction in order to add the supplier delay, the company
+        security lead time and the company days to purchase. The days to
+        purchase and company lead time are always displayed for onboarding
+        purpose in order to indicate that those confirmations are available.
+        """
+        lead_days, lead_days_description = super()._get_lead_days(product)
+        if 'buy' not in self.mapped('action') or not product._prepare_sellers():
+            return lead_days, lead_days_description
+        supplier_delay = product._prepare_sellers()[0].delay
+        if supplier_delay:
+            lead_days_description += '<tr><td>%s</td><td>+ %d %s</td></tr>' % (_('Vendor Lead Time'), supplier_delay, _('day(s)'))
+        security_delay = self.filtered(lambda r: r.action == 'buy').picking_type_id.company_id.po_lead
+        lead_days_description += '<tr><td>%s</td><td>+ %d %s</td></tr>' % (_('Purchase Security Lead Time'), security_delay, _('day(s)'))
+        days_to_purchase = self.filtered(lambda r: r.action == 'buy').company_id.days_to_purchase
```

Comment:
you're looking 3 times for the buy rule in this method, maybe you should store in a var like in mrp?

``` python
        delay, delay_description = super()._get_lead_days(product)
        buy_rule = self.filtered(lambda r: r.action == 'buy')
```